### PR TITLE
Reduce memory use and fix synchronization for dump resources

### PR DESCRIPTION
**Pull request by davidd-lunarg**
This PR builds on PR #11, fixing synchronization for dumping resources. The two changes are interrelated so they have been combined here into a single PR.

### CODE REVIEW

File: framework/graphics/dx12_dump_resources.h [hunk 1]
```diff
@@ -137,10 +148,7 @@ class Dx12DumpResources
     ~Dx12DumpResources();
 
     void StartDump(const TrackDumpResources& resources);
-    void WriteResource(const CopyResourceData&                             resource_data,
-                       const std::vector<std::pair<std::string, int32_t>>& json_path,
-                       const std::string&                                  file_name,
-                       const std::string&                                  type);
+    void WriteResource(const CopyResourceData* resource_data);
```
Comment (locke-lunarg) [inline framework/graphics/dx12_dump_resources.h]: Should we use `CopyResourceDataPtr` to replace all `CopyResourceData*`?

Comment (davidd-lunarg) [reply framework/graphics/dx12_dump_resources.h]: It isn't required here but I think that is a good idea for consistency. I will make that update.

Comment (davidd-lunarg) [reply framework/graphics/dx12_dump_resources.h]: Done

**Review by locke-lunarg — APPROVED** [review]
Thank you for fixing the sync issue.

**Pull request merged**